---
title: "Mahad-Bihie-RNA-Seq-Analysis"
output:
  html_document:
    toc: true
    toc_float: true
    theme: united
    code_folding: hide
---


```{r libraries, message=FALSE}
#Required Libraries

#BiocManager::install("DESeq2")
library(DESeq2)
#BiocManager::install("EnsDb.Hsapiens.v86")
library(EnsDb.Hsapiens.v86)
#BiocManager::install("EnhancedVolcano")
library(EnhancedVolcano)
#BiocManager::install("pheatmap")
library(pheatmap)
#BiocManager::install("org.Hs.eg.db")
library(org.Hs.eg.db)
#BiocManager::install("KEGGREST")
library(KEGGREST)
#BiocManager::install("pathview")
library(pathview)
#BiocManager::install("clusterProfiler")
library(clusterProfiler)
#BiocManager::install("gage")
library(gage)
#BiocManager::install("gageData")
library(gageData)
#install.packages("tidyverse")
library(tidyverse)
#install.packages("recount")
library(recount)
#install.packages("RColorBrewer")
library(RColorBrewer)
#install.packages("DT")
library(DT)
#install.packages("enrichR")
library(enrichR)
#install.packages("kableExtra")
library(kableExtra)
```


## Overview

Ewing sarcoma, a type of bone cancer common in children, is caused by the fusion of the Ewing Sarcoma Breakpoint Region 1 gene (EWSR1) and the Friend Leukemia Virus Integration 1 gene (FLI1). This results in a EWS-FLI1 fusion protein, which can then lead to oncogenesis of benign cells. Therefore, the EWSR1-FL1 fusion oncogene  is a therapeutic target for Ewing sarcoma. This project analyzes the impact that EWSR1-FLI1 suppressing therapies has on the transcriptome of Ewing sarcoma cells. 

To learn more about bioinformatics, I analyzed RNA-Seq data to find any potential genes associated with differences in the treated and control Ewing sarcoma samples. I processed a differential gene expression (DGE) analysis to confirm what genes were changed, when comparing samples with the EWSR1-FLI1 fusion gene silenced (`shEF1`) and the control samples (`shCTR`). I found the data and instructions on how to do this from the Bioinformatic Research Network.

## Construct the DESeq2 DataSet Object

I have created the DES object using the Ewing sarcoma RNA-seq data set provided. I have also made small changes to the DES object such as reducing the data to normalized values, removing low gene counts and removing the version number from the geneID.
```{r, message = FALSE, warning = FALSE}
#   class(rse): RangedSummarizedExperiment
rse <- readRDS("EwS.rds")

#remove the version number from the geneID
rownames(rse) <- gsub(rownames(rse),
                           pattern = "\\..+", replacement = "")

#make the dds
dds <- DESeqDataSet(rse, design = ~condition)

#set the factor level
dds$condition <- relevel(dds$condition, ref = "shCTR")

#creat DESeq 2 object
dds <- DESeq(dds)

#decrease the size of the DESeq object to make DESeq2 functions faster
dds <- estimateSizeFactors(dds)
idx <- rowSums( counts(dds, normalized=TRUE) >= 10 ) >= 3
dds <- dds[idx,]

#remove rows with low gene counts
keep <- as.data.frame(counts(dds)) %>%
  rowSums(counts(dds, normalized=TRUE)) >= 10
dds <- dds[keep,]
```

## 1. PCA Plot summarizing the sample-level variance within the data set.

A principal component analysis is used to emphasize the variation present in a dataset.This PCA plot displays the variation due to both the effect of the experiment and any batch effects as well. The control group and the experimental group are both clearly separated in this PCA with, with the first principal component dividing them with 98% of the variance.
```{r, message = FALSE}
#create rlog for PCA
rld <- rlog(dds)

#plot the PCA
plotadj <- plotPCA(rld)

#edit the plot height and width
plotadj + coord_fixed(ratio = 1,
                      xlim=c(-150, 150),
                      ylim=c(-100, 100))
```

## 2. MA Plot - Mean Expression VS Log 2 Fold Change

I first extracted the results and then shrunk the log fold change to remove any values where the fold change was exaggerated. The MA plot displays the relationship between the normalized mean expression and the log 2 fold change. The points shown at the edges of the y-axis (high absolute log fold change) are supposed to represent genes with high differential expression levels. As seen below, the only points with a high absolute log fold change also have high mean normalized counts. With the shrunken results, I was able to remove non-significant values with an extreme fold change to make a clearer graph that only presented differentially expressed values that are significant. There are also more significant genes with a positive log 2 fold change, indicating that there is a significant number of genes being upregulated. 

```{r, message = FALSE, warning = FALSE}
#extract the results
res <- results(dds,
               contrast = c("condition", 
                            "shCTR",      
                            "shEF1"),  
               alpha = 0.05)

#shrink data
resNorm <- lfcShrink(dds = dds,
                     res = res,
                     type = "normal",
                     coef = 2)

#MA Plot showing the relationship between mean count and log2 fold change.
plotMA(resNorm,
       main = "MA-plot of Normalized Ewing Sarcoma RNA-seq Data")
```

## 3. Table of Differentially Expressed Genes

I first used tidyverse code to align the data with the appropriate gene symbol. I then specified the leve of significance in the results and ordered the data according to the significance level. 
```{r, message = FALSE, warning = FALSE}

#make a dataframe of the results to view them
resdf <- as.data.frame(resNorm)

#extract gene symbol from EnsDb.Hsapiens.v86
ens2sym <- AnnotationDbi::select(EnsDb.Hsapiens.v86,
                                 keys = keys(EnsDb.Hsapiens.v86), 
                                 columns = c("SYMBOL"))

#join ens2sym to resdf by shared column (GENEID)
resdfsym <- resdf %>%
  rownames_to_column() %>%
  mutate(GENEID = gsub(rowname, pattern = "\\..+", replacement = "")) %>% 
  inner_join(y = ens2sym, by = "GENEID") %>% 
  dplyr::select(-rowname) %>%
  mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                          TRUE ~ padj)) #replacing 0s with minimum R value

#DF to specify level of signficance
DEgenes <- resdfsym %>%
  arrange(padj) %>%
  dplyr::filter(padj < 0.05 & log2FoldChange > 2) %>%  
  #slice_head(n=10) %>%
  select("SYMBOL", "GENEID", "padj", "baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue") 

#data table of DE genes
datatable(DEgenes,
          class = 'cell-border stripe',
          caption = htmltools::tags$caption(
          style = 'caption-side: bottom; text-align: center;',
                  'Table 1: ',
                  width = 3,
          htmltools::em('Differentially Expressed Genes.')),
          filter = 'top',
          extensions = 'FixedColumns',
          options = list(
              pageLength = 5,
              autoWidth = TRUE
              ,
              scrollX = TRUE
                        )
          )
```

## 4. Volcano Plot of all Differentially Expressed Genes


This plot visualizes genes with large fold changes that are also statistically significant. These may be the most biologically significant genes. In a volcano plot, the upregulated genes are towards the right, with the downregulated genes leaning towards the left, and the most statistically significant genes of both types of genes are at the top of the plot. 

This volcano plot presents a wide dispersion between the control and experimental group, which implies that there is a strong difference in gene expression. For example, COL11A1 is shown to have a small p-value and is therefore considered statistically differentially expressed. The are also many genes far from 0 which shows that are large differences in mean expression levels. Question 3 also supports the significance of COLL1A1.
```{r, message = FALSE, warning = FALSE, fig.height = 10, fig.width = 10}
#create the volcano plot
EnhancedVolcano(resdfsym, lab = resdfsym$SYMBOL, pCutoff = 0.05,
                FCcutoff = 2,
                x = "log2FoldChange",
                y = "padj",
                #xlim = c(-30, 30),
                title = "EWSR1-FLI1 Suppressed Ewing Sarcoma DEGs")
```

## 5. Heatmap of Top 10 Over Expressed & Under expressed DEGs
The heatmap below shows a clear distinction between the untreated and treated samples in terms of gene expression. Under the treated columns, you can see which genes are over-expressed and under-expressed as they are coloured according to the gradient on the right. The untreated samples are down regulated for the top 10 over-expressed genes shown, with mixed results for the bottom 10 under expressed genes. 
```{r, message = FALSE}
#filter df to significant DEGs that are both over expressed and underexpressed
orderedSig <-resdfsym %>%
  arrange(padj) %>%
  dplyr::filter(padj < 0.05 & log2FoldChange > 2 | log2FoldChange < -2) 

#create matrix for heat map
id1 <- orderedSig$GENEID 
id2 <- orderedSig$SYMBOL 
mat <- assay(rld) #extract mtx of nrmlzd counts
DE <- mat[id1,]
rownames(DE) <- id2 #rmv ensemble iDs from row name

#transform matrix to df to filter to top 10 DEGs
#overexpressed
overDE <- as.data.frame(DE) %>% #filter to top 10 overexpressed genes
  slice_head(n=10)
overDE <- as.matrix(overDE) #make it a matrix for pheatmap
#underexpressed
underDE <- as.data.frame(DE) %>% #filter to top 10 underrexpressed genes
  slice_tail(n=10)
underDE <- as.matrix(underDE) #make it a matrix for pheatmap


DE <- rbind(overDE, underDE) #bind both matrices into 1
annotation <- as.data.frame(colData(rld)[, c("run","condition")]) #add annotation onto map

#heatmap
pheatmap(DE,
         scale = "row", 
         clustering_distance_rows = "correlation", 
         cluster_rows=F, 
         cluster_cols=F,
         annotation_col = annotation,
         show_colnames = FALSE, # removes coloumn names from run at the btm
         annotation_names_col = FALSE, #removes names beside the colours at the top
         main="Top 10 Differentially Expressed genes"
)
```


## Conclusion
The tools used above imply that there is a statistically significant difference in gene expression between the samples that treat the EWSR1-FL1 fusion oncogene and the untreated samples. This implies that there is a change in the transcriptome of Ewing sarcoma cells when the proposed source of this cancer is targeted. There were also few genes shown to have high upregulation in the treated group, with COL11A1 for example shown at the top of the list in sections 3, 4 and 5. This project visualized the genes that are relevant in the  suppression of the EWSR1-FL1 fusion oncogene in Ewing sarcoma cells. One way to expand on this would be to visualize gene pathways to provide more insight on the effects of this treatment.
